ga4gh/backend.py

```diff
@@ -395,8 +403,21 @@ def referenceSetsGenerator(self, request):
         Returns a generator over the (referenceSet, nextPageToken) pairs
         defined by the specified request.
         """
-        return self._topLevelObjectGenerator(
-            request, self.getNumReferenceSets(), self.getReferenceSetByIndex)
+        results = []
+        for obj in self.getReferenceSets():
+            include = True
+            if request.md5checksum is not None:
+                if request.md5checksum != obj.getMd5Checksum():
+                    include = False
+            if request.accession is not None:
```

Comment:
I think it's easier to read as it is --- the time penalty will be trivial I think. I can add
```python
if not include and request.accession is not None:
    # stuff
``` 
if you prefer it though, I don't have a strong opinion.